diff --git a/clone/modeling_roberta.py b/clone/modeling_roberta.py
index 092cda9..59311b1 100644
--- a/clone/modeling_roberta.py
+++ b/clone/modeling_roberta.py
@@ -1652,7 +1652,7 @@ class RobertaForQuestionAnswering(RobertaPreTrainedModel):
 
     @add_start_docstrings_to_model_forward(ROBERTA_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
     @add_code_sample_docstrings(
-        tokenizer_class=_TOKENIZER_FOR_DOC,
+        # tokenizer_class=_TOKENIZER_FOR_DOC,
         checkpoint=_CHECKPOINT_FOR_DOC,
         output_type=QuestionAnsweringModelOutput,
         config_class=_CONFIG_FOR_DOC,
diff --git a/clone/run-MHM.sh b/clone/run-MHM.sh
deleted file mode 100644
index c0c280d..0000000
--- a/clone/run-MHM.sh
+++ /dev/null
@@ -1,159 +0,0 @@
-#! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
-export HF_DATASETS_CACHE=checkpoints/hf_model
-export HF_METRICS_CACHE=checkpoints/hf_model
-
-cache_dir=${TRANSFORMERS_CACHE}
-
-
-metric="accuracy"
-export WANDB_PROJECT=glue
-export WANDB_WATCH="false"
-
-seed=42
-
-# ----- MAM adapter -----
-attn_mode="prefix"
-attn_option="concat"
-attn_composition="add"
-attn_bn=16  # attn bottleneck dim
-
-ffn_mode="adapter"
-ffn_option="sequential"
-ffn_adapter_layernorm_option="none"
-ffn_adapter_init_option="lora"
-ffn_adapter_scalar="2"
-ffn_bn=16 # ffn bottleneck dim
-
-# ----- lora -----
-# attn_mode="lora"
-# attn_option="none"
-# attn_composition="add"
-# attn_bn=16
-
-# set ffn_mode to be 'lora' to use
-# lora at ffn as well
-
-# ffn_mode="lora"
-# ffn_option="none"
-# ffn_adapter_layernorm_option="none"
-# ffn_adapter_init_option="bert"
-# ffn_adapter_scalar="1"
-# ffn_bn=16
-
-lora_alpha=32
-lora_dropout=0.1
-lora_init="lora"
-
-
-# lora params are not set
-if [ -z ${lora_alpha+x} ];
-then
-    lora_alpha=0
-    lora_init="lora"
-    lora_dropout=0
-fi
-
-# set to 1 for debug mode which only
-# uses 1600 training examples
-debug=0
-
-# set to "wandb" to use weights & bias
-report_to="none"
-
-bsz=40
-gradient_steps=1
-
-# lr=5e-5
-lr=5e-5
-max_grad_norm=1
-# lr=1e-5
-# weight_decay=0
-weight_decay=0.1
-warmup_updates=0
-warmup_ratio=0.06
-max_steps=-1
-num_train_epochs=6
-max_tokens_per_batch=0
-max_seq_length=512
-
-lr_scheduler_type="polynomial"
-#metric=bleu
-unfreeze='ef_'
-max_eval_samples=1600
-logging_steps=1000
-
-eval_strategy="epoch"
-save_steps=5000
-
-# for seed in "${seed_list[@]}"; do
-model_name='codebert'
-exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
-exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
-exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
-exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
-exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}
-SAVE=checkpoints/${exp_name}
-
-if [ ! -d ${SAVE} ];then
-    mkdir -p ${SAVE}
-fi
-CUDA_VISIBLE_DEVICES=0 python -u run.py \
-    --model_name_or_path microsoft/codebert-base \
-    --do_train \
-    --do_eval \
-    --do_predict \
-    --max_seq_length 512 \
-    --per_device_train_batch_size ${bsz} \
-    --per_device_eval_batch_size ${bsz} \
-    --max_tokens_per_batch ${max_tokens_per_batch} \
-    --adam_beta1 0.9 \
-    --adam_beta2 0.98 \
-    --adam_epsilon 1e-6 \
-    --lora_alpha ${lora_alpha} \
-    --lora_dropout ${lora_dropout} \
-    --lora_init ${lora_init} \
-    --attn_mode ${attn_mode} \
-    --attn_option ${attn_option} \
-    --attn_composition ${attn_composition} \
-    --ffn_mode ${ffn_mode} \
-    --ffn_option ${ffn_option} \
-    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
-    --ffn_adapter_scalar ${ffn_adapter_scalar} \
-    --ffn_adapter_init_option ${ffn_adapter_init_option} \
-    --mid_dim 800 \
-    --attn_bn ${attn_bn} \
-    --ffn_bn ${ffn_bn} \
-    --seed ${seed} \
-    --unfreeze_params ${unfreeze} \
-    --max_eval_samples ${max_eval_samples} \
-    --gradient_accumulation_steps ${gradient_steps} \
-    --max_steps ${max_steps} \
-    --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train_sampled.txt \
-    --valid_filename=dataset/valid_sampled.txt \
-    --test_filename=dataset/test_sampled.txt \
-    --learning_rate ${lr} \
-    --lr_scheduler_type ${lr_scheduler_type} \
-    --max_grad_norm ${max_grad_norm} \
-    --weight_decay ${weight_decay} \
-    --warmup_steps ${warmup_updates} \
-    --warmup_ratio ${warmup_ratio} \
-    --max_seq_length ${max_seq_length} \
-    --fp16 \
-    --logging_steps ${logging_steps} \
-    --save_total_limit 1 \
-    --evaluation_strategy ${eval_strategy} \
-    --save_strategy ${eval_strategy} \
-    --save_steps ${save_steps} \
-    --eval_steps ${save_steps} \
-    --load_best_model_at_end \
-    --report_to ${report_to} \
-    --run_name ${exp_name} \
-    --overwrite_output_dir \
-    --metric_for_best_model ${metric} \
-    --greater_is_better "True" \
-    --ddp_find_unused_parameter "False" \
-    --output_dir ${SAVE} \
-        2>&1 | tee ${SAVE}/log.txt
diff --git a/clone/run-adapter.sh b/clone/run-adapter.sh
deleted file mode 100644
index 20e7388..0000000
--- a/clone/run-adapter.sh
+++ /dev/null
@@ -1,159 +0,0 @@
-#! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
-export HF_DATASETS_CACHE=checkpoints/hf_model
-export HF_METRICS_CACHE=checkpoints/hf_model
-
-cache_dir=${TRANSFORMERS_CACHE}
-
-
-metric="accuracy"
-export WANDB_PROJECT=glue
-export WANDB_WATCH="false"
-
-seed=42
-
-# ----- MAM adapter -----
-attn_mode="none"
-attn_option="none"
-attn_composition="add"
-attn_bn=16  # attn bottleneck dim
-
-ffn_mode="adapter"
-ffn_option="sequential"
-ffn_adapter_layernorm_option="none"
-ffn_adapter_init_option="lora"
-ffn_adapter_scalar="2"
-ffn_bn=16 # ffn bottleneck dim
-
-# ----- lora -----
-# attn_mode="lora"
-# attn_option="none"
-# attn_composition="add"
-# attn_bn=16
-
-# set ffn_mode to be 'lora' to use
-# lora at ffn as well
-
-# ffn_mode="lora"
-# ffn_option="none"
-# ffn_adapter_layernorm_option="none"
-# ffn_adapter_init_option="bert"
-# ffn_adapter_scalar="1"
-# ffn_bn=16
-
-lora_alpha=32
-lora_dropout=0.1
-lora_init="lora"
-
-
-# lora params are not set
-if [ -z ${lora_alpha+x} ];
-then
-    lora_alpha=0
-    lora_init="lora"
-    lora_dropout=0
-fi
-
-# set to 1 for debug mode which only
-# uses 1600 training examples
-debug=0
-
-# set to "wandb" to use weights & bias
-report_to="none"
-
-bsz=40
-gradient_steps=1
-
-# lr=5e-5
-lr=5e-5
-max_grad_norm=1
-# lr=1e-5
-# weight_decay=0
-weight_decay=0.1
-warmup_updates=0
-warmup_ratio=0.06
-max_steps=-1
-num_train_epochs=6
-max_tokens_per_batch=0
-max_seq_length=512
-
-lr_scheduler_type="polynomial"
-#metric=bleu
-unfreeze='ef_'
-max_eval_samples=1600
-logging_steps=1000
-
-eval_strategy="epoch"
-save_steps=5000
-
-# for seed in "${seed_list[@]}"; do
-model_name='codebert'
-exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
-exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
-exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
-exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
-exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}
-SAVE=checkpoints/${exp_name}
-
-if [ ! -d ${SAVE} ];then
-    mkdir -p ${SAVE}
-fi
-CUDA_VISIBLE_DEVICES=0 python -u run.py \
-    --model_name_or_path microsoft/codebert-baset \
-    --do_train \
-    --do_eval \
-    --do_predict \
-    --max_seq_length 512 \
-    --per_device_train_batch_size ${bsz} \
-    --per_device_eval_batch_size ${bsz} \
-    --max_tokens_per_batch ${max_tokens_per_batch} \
-    --adam_beta1 0.9 \
-    --adam_beta2 0.98 \
-    --adam_epsilon 1e-6 \
-    --lora_alpha ${lora_alpha} \
-    --lora_dropout ${lora_dropout} \
-    --lora_init ${lora_init} \
-    --attn_mode ${attn_mode} \
-    --attn_option ${attn_option} \
-    --attn_composition ${attn_composition} \
-    --ffn_mode ${ffn_mode} \
-    --ffn_option ${ffn_option} \
-    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
-    --ffn_adapter_scalar ${ffn_adapter_scalar} \
-    --ffn_adapter_init_option ${ffn_adapter_init_option} \
-    --mid_dim 800 \
-    --attn_bn ${attn_bn} \
-    --ffn_bn ${ffn_bn} \
-    --seed ${seed} \
-    --unfreeze_params ${unfreeze} \
-    --max_eval_samples ${max_eval_samples} \
-    --gradient_accumulation_steps ${gradient_steps} \
-    --max_steps ${max_steps} \
-    --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train_sampled.txt \
-    --valid_filename=dataset/valid_sampled.txt \
-    --test_filename=dataset/test_sampled.txt \
-    --learning_rate ${lr} \
-    --lr_scheduler_type ${lr_scheduler_type} \
-    --max_grad_norm ${max_grad_norm} \
-    --weight_decay ${weight_decay} \
-    --warmup_steps ${warmup_updates} \
-    --warmup_ratio ${warmup_ratio} \
-    --max_seq_length ${max_seq_length} \
-    --fp16 \
-    --logging_steps ${logging_steps} \
-    --save_total_limit 1 \
-    --evaluation_strategy ${eval_strategy} \
-    --save_strategy ${eval_strategy} \
-    --save_steps ${save_steps} \
-    --eval_steps ${save_steps} \
-    --load_best_model_at_end \
-    --report_to ${report_to} \
-    --run_name ${exp_name} \
-    --overwrite_output_dir \
-    --metric_for_best_model ${metric} \
-    --greater_is_better "True" \
-    --ddp_find_unused_parameter "False" \
-    --output_dir ${SAVE} \
-        2>&1 | tee ${SAVE}/log.txt
diff --git a/clone/run-lora.sh b/clone/run-lora.sh
index 80457bb..fc08ad9 100644
--- a/clone/run-lora.sh
+++ b/clone/run-lora.sh
@@ -1,10 +1,22 @@
 #! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
+#SBATCH --time=15:00:00          # Maximum run time in hh:mm:ss
+#SBATCH --mem=32000             # Maximum memory required (in megabytes)
+#SBATCH --job-name=default_479  # Job name (to track progress)
+#SBATCH --partition=gpu         # Partition on which to run job
+#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
+#SBATCH --constraint=gpu_v100
+#SBATCH --licenses=common
+
+module load anaconda
+conda activate fine-tune-bert
+
+export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
+export TF_ENABLE_ONEDNN_OPTS=0
+export HF_HOME=checkpoints/hf_model
 export HF_DATASETS_CACHE=checkpoints/hf_model
 export HF_METRICS_CACHE=checkpoints/hf_model
 
-cache_dir=${TRANSFORMERS_CACHE}
+cache_dir=${HF_HOME}
 
 
 metric="f1"
@@ -74,7 +86,7 @@ weight_decay=0.1
 warmup_updates=0
 warmup_ratio=0.06
 max_steps=-1
-num_train_epochs=15
+num_train_epochs=10
 max_tokens_per_batch=0
 max_seq_length=512
 
@@ -82,6 +94,7 @@ lr_scheduler_type="polynomial"
 #metric=bleu
 unfreeze='ef_'
 max_eval_samples=1600
+max_train_samples=100
 logging_steps=1000
 
 eval_strategy="epoch"
@@ -101,7 +114,7 @@ if [ ! -d ${SAVE} ];then
 fi
 CUDA_VISIBLE_DEVICES=0 python -u run.py \
     --model_name_or_path microsoft/codebert-base \
-    --do_train False\
+    --do_train \
     --do_eval \
     --do_predict \
     --max_seq_length 512 \
@@ -127,13 +140,14 @@ CUDA_VISIBLE_DEVICES=0 python -u run.py \
     --ffn_bn ${ffn_bn} \
     --seed ${seed} \
     --unfreeze_params ${unfreeze} \
+    --max_train_samples ${max_train_samples} \
     --max_eval_samples ${max_eval_samples} \
     --gradient_accumulation_steps ${gradient_steps} \
     --max_steps ${max_steps} \
     --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train_sampled.txt \
-    --valid_filename=dataset/valid_sampled.txt \
-    --test_filename=dataset/test_sampled.txt \
+    --train_filename=dataset/train.txt \
+    --valid_filename=dataset/valid.txt \
+    --test_filename=dataset/test.txt \
     --learning_rate ${lr} \
     --lr_scheduler_type ${lr_scheduler_type} \
     --max_grad_norm ${max_grad_norm} \
diff --git a/clone/run-none.sh b/clone/run-none.sh
index 65bd7b6..68718d5 100644
--- a/clone/run-none.sh
+++ b/clone/run-none.sh
@@ -1,10 +1,22 @@
 #! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
+#SBATCH --time=12:00:00          # Maximum run time in hh:mm:ss
+#SBATCH --mem=32000             # Maximum memory required (in megabytes)
+#SBATCH --job-name=default_479  # Job name (to track progress)
+#SBATCH --partition=gpu         # Partition on which to run job
+#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
+#SBATCH --constraint='gpu_32gb&gpu_v100'
+#SBATCH --licenses=common
+
+module load anaconda
+conda activate fine-tune-bert
+
+export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
+export TF_ENABLE_ONEDNN_OPTS=0
+export HF_HOME=checkpoints/hf_model
 export HF_DATASETS_CACHE=checkpoints/hf_model
 export HF_METRICS_CACHE=checkpoints/hf_model
 
-cache_dir=${TRANSFORMERS_CACHE}
+cache_dir=${HF_HOME}
 
 
 metric="accuracy"
@@ -16,7 +28,7 @@ seed=42
 # ----- MAM adapter -----
 attn_mode="none"
 attn_option="none"
-attn_composition="none"
+attn_composition="add"
 attn_bn=16  # attn bottleneck dim
 
 ffn_mode="none"
@@ -74,7 +86,7 @@ weight_decay=0.1
 warmup_updates=0
 warmup_ratio=0.06
 max_steps=-1
-num_train_epochs=6
+num_train_epochs=5
 max_tokens_per_batch=0
 max_seq_length=512
 
@@ -82,6 +94,7 @@ lr_scheduler_type="polynomial"
 #metric=bleu
 unfreeze='ef_'
 max_eval_samples=1600
+max_train_samples=100
 logging_steps=1000
 
 eval_strategy="epoch"
@@ -127,13 +140,14 @@ CUDA_VISIBLE_DEVICES=0 python -u run.py \
     --ffn_bn ${ffn_bn} \
     --seed ${seed} \
     --unfreeze_params ${unfreeze} \
+    --max_train_samples ${max_train_samples} \
     --max_eval_samples ${max_eval_samples} \
     --gradient_accumulation_steps ${gradient_steps} \
     --max_steps ${max_steps} \
     --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train_sampled.txt \
-    --valid_filename=dataset/valid_sampled.txt \
-    --test_filename=dataset/test_sampled.txt \
+    --train_filename=dataset/train.txt \
+    --valid_filename=dataset/valid.txt \
+    --test_filename=dataset/test.txt \
     --learning_rate ${lr} \
     --lr_scheduler_type ${lr_scheduler_type} \
     --max_grad_norm ${max_grad_norm} \
diff --git a/clone/run-parallel-adapter.sh b/clone/run-parallel-adapter.sh
index ab91f4d..d3509c5 100644
--- a/clone/run-parallel-adapter.sh
+++ b/clone/run-parallel-adapter.sh
@@ -1,10 +1,22 @@
 #! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
+#SBATCH --time=15:00:00          # Maximum run time in hh:mm:ss
+#SBATCH --mem=32000             # Maximum memory required (in megabytes)
+#SBATCH --job-name=default_479  # Job name (to track progress)
+#SBATCH --partition=gpu         # Partition on which to run job
+#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
+#SBATCH --constraint='gpu_32gb&gpu_v100'
+#SBATCH --licenses=common
+
+module load anaconda
+conda activate fine-tune-bert
+
+export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
+export TF_ENABLE_ONEDNN_OPTS=0
+export HF_HOME=checkpoints/hf_model
 export HF_DATASETS_CACHE=checkpoints/hf_model
 export HF_METRICS_CACHE=checkpoints/hf_model
 
-cache_dir=${TRANSFORMERS_CACHE}
+cache_dir=${HF_HOME}
 
 
 metric="accuracy"
@@ -16,7 +28,7 @@ seed=42
 # ----- MAM adapter -----
 attn_mode="none"
 attn_option="none"
-attn_composition="none"
+attn_composition="add"
 attn_bn=16  # attn bottleneck dim
 
 ffn_mode="adapter"
@@ -74,7 +86,7 @@ weight_decay=0.1
 warmup_updates=0
 warmup_ratio=0.06
 max_steps=-1
-num_train_epochs=6
+num_train_epochs=10
 max_tokens_per_batch=0
 max_seq_length=512
 
@@ -82,6 +94,7 @@ lr_scheduler_type="polynomial"
 #metric=bleu
 unfreeze='ef_'
 max_eval_samples=1600
+max_train_samples=100
 logging_steps=1000
 
 eval_strategy="epoch"
@@ -127,13 +140,14 @@ CUDA_VISIBLE_DEVICES=0 python -u run.py \
     --ffn_bn ${ffn_bn} \
     --seed ${seed} \
     --unfreeze_params ${unfreeze} \
+    --max_train_samples ${max_train_samples} \
     --max_eval_samples ${max_eval_samples} \
     --gradient_accumulation_steps ${gradient_steps} \
     --max_steps ${max_steps} \
     --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train_sampled.txt \
-    --valid_filename=dataset/valid_sampled.txt \
-    --test_filename=dataset/test_sampled.txt \
+    --train_filename=dataset/train.txt \
+    --valid_filename=dataset/valid.txt \
+    --test_filename=dataset/test.txt \
     --learning_rate ${lr} \
     --lr_scheduler_type ${lr_scheduler_type} \
     --max_grad_norm ${max_grad_norm} \
diff --git a/clone/run-prefix.sh b/clone/run-prefix.sh
deleted file mode 100644
index 028eb3b..0000000
--- a/clone/run-prefix.sh
+++ /dev/null
@@ -1,159 +0,0 @@
-#! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
-export HF_DATASETS_CACHE=checkpoints/hf_model
-export HF_METRICS_CACHE=checkpoints/hf_model
-
-cache_dir=${TRANSFORMERS_CACHE}
-
-
-metric="accuracy"
-export WANDB_PROJECT=glue
-export WANDB_WATCH="false"
-
-seed=42
-
-# ----- MAM adapter -----
-attn_mode="prefix"
-attn_option="concat"
-attn_composition="add"
-attn_bn=16  # attn bottleneck dim
-
-ffn_mode="none"
-ffn_option="none"
-ffn_adapter_layernorm_option="none"
-ffn_adapter_init_option="lora"
-ffn_adapter_scalar="2"
-ffn_bn=16 # ffn bottleneck dim
-
-# ----- lora -----
-# attn_mode="lora"
-# attn_option="none"
-# attn_composition="add"
-# attn_bn=16
-
-# set ffn_mode to be 'lora' to use
-# lora at ffn as well
-
-# ffn_mode="lora"
-# ffn_option="none"
-# ffn_adapter_layernorm_option="none"
-# ffn_adapter_init_option="bert"
-# ffn_adapter_scalar="1"
-# ffn_bn=16
-
-# lora_alpha=32
-# lora_dropout=0.1
-# lora_init="lora"
-
-
-# lora params are not set
-if [ -z ${lora_alpha+x} ];
-then
-    lora_alpha=0
-    lora_init="lora"
-    lora_dropout=0
-fi
-
-# set to 1 for debug mode which only
-# uses 1600 training examples
-debug=0
-
-# set to "wandb" to use weights & bias
-report_to="none"
-
-bsz=40
-gradient_steps=1
-
-# lr=5e-5
-lr=5e-5
-max_grad_norm=1
-# lr=1e-5
-# weight_decay=0
-weight_decay=0.1
-warmup_updates=0
-warmup_ratio=0.06
-max_steps=-1
-num_train_epochs=6
-max_tokens_per_batch=0
-max_seq_length=512
-
-lr_scheduler_type="polynomial"
-#metric=bleu
-unfreeze='ef_'
-max_eval_samples=1600
-logging_steps=1000
-
-eval_strategy="epoch"
-save_steps=5000
-
-# for seed in "${seed_list[@]}"; do
-model_name='codebert'
-exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
-exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
-exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
-exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
-exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}_tes4
-SAVE=checkpoints/${exp_name}
-
-if [ ! -d ${SAVE} ];then
-    mkdir -p ${SAVE}
-fi
-CUDA_VISIBLE_DEVICES=0 python -u run.py \
-    --model_name_or_path microsoft/codebert-base \
-    --do_train \
-    --do_eval \
-    --do_predict \
-    --max_seq_length 512 \
-    --per_device_train_batch_size ${bsz} \
-    --per_device_eval_batch_size ${bsz} \
-    --max_tokens_per_batch ${max_tokens_per_batch} \
-    --adam_beta1 0.9 \
-    --adam_beta2 0.98 \
-    --adam_epsilon 1e-6 \
-    --lora_alpha ${lora_alpha} \
-    --lora_dropout ${lora_dropout} \
-    --lora_init ${lora_init} \
-    --attn_mode ${attn_mode} \
-    --attn_option ${attn_option} \
-    --attn_composition ${attn_composition} \
-    --ffn_mode ${ffn_mode} \
-    --ffn_option ${ffn_option} \
-    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
-    --ffn_adapter_scalar ${ffn_adapter_scalar} \
-    --ffn_adapter_init_option ${ffn_adapter_init_option} \
-    --mid_dim 400 \
-    --attn_bn ${attn_bn} \
-    --ffn_bn ${ffn_bn} \
-    --seed ${seed} \
-    --unfreeze_params ${unfreeze} \
-    --max_eval_samples ${max_eval_samples} \
-    --gradient_accumulation_steps ${gradient_steps} \
-    --max_steps ${max_steps} \
-    --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train_sampled.txt \
-    --valid_filename=dataset/valid_sampled.txt \
-    --test_filename=dataset/test_sampled.txt \
-    --learning_rate ${lr} \
-    --lr_scheduler_type ${lr_scheduler_type} \
-    --max_grad_norm ${max_grad_norm} \
-    --weight_decay ${weight_decay} \
-    --warmup_steps ${warmup_updates} \
-    --warmup_ratio ${warmup_ratio} \
-    --max_seq_length ${max_seq_length} \
-    --fp16 \
-    --logging_steps ${logging_steps} \
-    --save_total_limit 1 \
-    --evaluation_strategy ${eval_strategy} \
-    --save_strategy ${eval_strategy} \
-    --save_steps ${save_steps} \
-    --eval_steps ${save_steps} \
-    --load_best_model_at_end \
-    --report_to ${report_to} \
-    --run_name ${exp_name} \
-    --overwrite_output_dir \
-    --metric_for_best_model ${metric} \
-    --greater_is_better "True" \
-    --ddp_find_unused_parameter "False" \
-    --output_dir ${SAVE} \
-        2>&1 | tee ${SAVE}/log.txt
diff --git a/clone/run.py b/clone/run.py
index 4e60029..cff9621 100644
--- a/clone/run.py
+++ b/clone/run.py
@@ -24,6 +24,7 @@ import numpy as np
 from utils import CodeDataset, get_last_checkpoint
 from sklearn.metrics import recall_score, precision_score, f1_score
 import transformers
+from transformers import BertConfig
 from transformers import (
     AutoTokenizer,
     HfArgumentParser,
@@ -169,7 +170,12 @@ def main():
     parser = HfArgumentParser((ModelArguments, DataTrainingArguments, TrainingArguments,
                                TuneArguments))
     model_args, data_args, training_args, tune_args = parser.parse_args_into_dataclasses()
-
+    print("Model Path:", model_args.model_name_or_path)
+    print("Config Name:", model_args.config_name)
+    print("Tokenizer Name:", model_args.tokenizer_name)
+    print("Use Auth Token:", model_args.use_auth_token)
+    # model_args.use_auth_token = "Bearer <REDACTED-HF-TOKEN>"
+    print("Use Auth Token:", model_args.use_auth_token)
     # Setup logging
     logging.basicConfig(
         format="%(asctime)s - %(levelname)s - %(name)s -   %(message)s",
@@ -191,19 +197,41 @@ def main():
 
     logger.info(f"Training/evaluation parameters {training_args}")
 
-    # Set seed before initializing model.
+    # # Set seed before initializing model.
     set_seed(training_args.seed)
     last_checkpoint = get_last_checkpoint(training_args.output_dir)
-    config = RobertaConfig.from_pretrained(
-        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+    if last_checkpoint:
+        logger.info(f"Resuming from last checkpoint: {last_checkpoint}")
+    else:
+        logger.info("Starting training from scratch")
+    # print("cheackpoint:", last_checkpoint)
+    # config = RobertaConfig.from_pretrained(
+    #    'roberta-base',
+    #     num_labels=2
+    # )
+    # if training_args.do_predict and not training_args.do_train:
+    #     print('hereeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee')
+    #     config = RobertaConfig.from_pretrained(
+    #         'roberta-base',
+    #         last_checkpoint,
+    #         num_labels=2
+    #     )
+    print("Checkpoint:", last_checkpoint)
+    config = BertConfig.from_pretrained(
+        'microsoft/codebert-base',
         num_labels=2
     )
+
     if training_args.do_predict and not training_args.do_train:
-        config = RobertaConfig.from_pretrained(
+        print('Configuration for prediction...')
+        config = BertConfig.from_pretrained(
+            'microsoft/codebert-base',
             last_checkpoint,
             num_labels=2
         )
 
+
+
     # put useful args into config: these arguments will be used in models, thus adding them to config
     # interested_args = ['use_prefix', 'mid_dim', 'preseqlen', 'prefix_dropout', 'unfreeze_params']
     for k, v in vars(tune_args).items():
@@ -218,6 +246,7 @@ def main():
         use_fast=model_args.use_fast_tokenizer,
         revision=model_args.model_revision,
         use_auth_token=True if model_args.use_auth_token else None,
+        token="Bearer <REDACTED-HF-TOKEN>"
     )
     model = RobertaForSequenceClassification.from_pretrained(
         model_args.model_name_or_path,
@@ -226,8 +255,11 @@ def main():
         cache_dir=model_args.cache_dir,
         revision=model_args.model_revision,
         use_auth_token=True if model_args.use_auth_token else None,
+        token="Bearer <REDACTED-HF-TOKEN>"
     )
-
+    
+    # if last_checkpoint:
+    #     model.load_state_dict(torch.load(os.path.join(last_checkpoint, "pytorch_model.bin")))
     dataset_loader = CodeDataset(os.path.dirname(model_args.train_filename))
 
     label_list = [0, 1]
@@ -272,6 +304,10 @@ def main():
     if training_args.do_train:
         train_dataset = dataset_loader.load(model_args.train_filename, model_args.data_num)
         eval_dataset = dataset_loader.load(model_args.valid_filename, model_args.data_num)
+        
+        if data_args.max_train_samples is not None:
+            train_dataset = train_dataset.select(range(min(data_args.max_train_samples, len(train_dataset))))
+
         train_dataset = train_dataset.map(
             preprocess_function,
             batched=True,
@@ -293,15 +329,17 @@ def main():
             data_collator=default_data_collator,
         )
 
-        train_result = trainer.train()
+        # train_result = trainer.train()
+        train_result = trainer.train(resume_from_checkpoint=last_checkpoint if last_checkpoint else None)
         metrics = train_result.metrics
         max_train_samples = (
             data_args.max_train_samples if data_args.max_train_samples is not None else len(train_dataset)
         )
         metrics["train_samples"] = min(max_train_samples, len(train_dataset))
 
-        trainer.save_model()  # Saves the tokenizer too for easy upload
-
+        trainer.save_model(output_dir=training_args.output_dir)  # Saves the tokenizer too for easy upload
+        tokenizer.save_pretrained(training_args.output_dir)
+        
         trainer.log_metrics("train", metrics)
         trainer.save_metrics("train", metrics)
         trainer.save_state()
@@ -317,7 +355,7 @@ def main():
         if not training_args.do_train:
 
             logger.info("load model state from %s", last_checkpoint)
-            model.load_state_dict(torch.load(os.path.join(last_checkpoint, "pytorch_model.bin")))
+            # model.load_state_dict(torch.load(os.path.join(last_checkpoint, "pytorch_model.bin")))
             # Initialize our Trainer
             trainer = Trainer(
                 model=model,
diff --git a/defect/run-MHM.sh b/defect/run-MHM.sh
deleted file mode 100644
index f53928b..0000000
--- a/defect/run-MHM.sh
+++ /dev/null
@@ -1,159 +0,0 @@
-#! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
-export HF_DATASETS_CACHE=checkpoints/hf_model
-export HF_METRICS_CACHE=checkpoints/hf_model
-
-cache_dir=${TRANSFORMERS_CACHE}
-
-
-metric="accuracy"
-export WANDB_PROJECT=glue
-export WANDB_WATCH="false"
-
-seed=42
-
-# ----- MAM adapter -----
-attn_mode="prefix"
-attn_option="concat"
-attn_composition="add"
-attn_bn=16  # attn bottleneck dim
-
-ffn_mode="adapter"
-ffn_option="sequential"
-ffn_adapter_layernorm_option="none"
-ffn_adapter_init_option="lora"
-ffn_adapter_scalar="2"
-ffn_bn=16 # ffn bottleneck dim
-
-# ----- lora -----
-# attn_mode="lora"
-# attn_option="none"
-# attn_composition="add"
-# attn_bn=16
-
-# set ffn_mode to be 'lora' to use
-# lora at ffn as well
-
-# ffn_mode="lora"
-# ffn_option="none"
-# ffn_adapter_layernorm_option="none"
-# ffn_adapter_init_option="bert"
-# ffn_adapter_scalar="1"
-# ffn_bn=16
-
-lora_alpha=32
-lora_dropout=0.1
-lora_init="lora"
-
-
-# lora params are not set
-if [ -z ${lora_alpha+x} ];
-then
-    lora_alpha=0
-    lora_init="lora"
-    lora_dropout=0
-fi
-
-# set to 1 for debug mode which only
-# uses 1600 training examples
-debug=0
-
-# set to "wandb" to use weights & bias
-report_to="none"
-
-bsz=40
-gradient_steps=1
-
-# lr=5e-5
-lr=5e-5
-max_grad_norm=1
-# lr=1e-5
-# weight_decay=0
-weight_decay=0.1
-warmup_updates=0
-warmup_ratio=0.06
-max_steps=-1
-num_train_epochs=6
-max_tokens_per_batch=0
-max_seq_length=512
-
-lr_scheduler_type="polynomial"
-#metric=bleu
-unfreeze='ef_'
-max_eval_samples=1600
-logging_steps=1000
-
-eval_strategy="epoch"
-save_steps=5000
-
-# for seed in "${seed_list[@]}"; do
-model_name='codebert'
-exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
-exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
-exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
-exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
-exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}
-SAVE=checkpoints/${exp_name}
-
-if [ ! -d ${SAVE} ];then
-    mkdir -p ${SAVE}
-fi
-CUDA_VISIBLE_DEVICES=0 python -u run.py \
-    --model_name_or_path microsoft/codebert-base \
-    --do_train \
-    --do_eval \
-    --do_predict \
-    --max_seq_length 512 \
-    --per_device_train_batch_size ${bsz} \
-    --per_device_eval_batch_size ${bsz} \
-    --max_tokens_per_batch ${max_tokens_per_batch} \
-    --adam_beta1 0.9 \
-    --adam_beta2 0.98 \
-    --adam_epsilon 1e-6 \
-    --lora_alpha ${lora_alpha} \
-    --lora_dropout ${lora_dropout} \
-    --lora_init ${lora_init} \
-    --attn_mode ${attn_mode} \
-    --attn_option ${attn_option} \
-    --attn_composition ${attn_composition} \
-    --ffn_mode ${ffn_mode} \
-    --ffn_option ${ffn_option} \
-    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
-    --ffn_adapter_scalar ${ffn_adapter_scalar} \
-    --ffn_adapter_init_option ${ffn_adapter_init_option} \
-    --mid_dim 800 \
-    --attn_bn ${attn_bn} \
-    --ffn_bn ${ffn_bn} \
-    --seed ${seed} \
-    --unfreeze_params ${unfreeze} \
-    --max_eval_samples ${max_eval_samples} \
-    --gradient_accumulation_steps ${gradient_steps} \
-    --max_steps ${max_steps} \
-    --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train.jsonl \
-    --valid_filename=dataset/valid.jsonl \
-    --test_filename=dataset/test.jsonl \
-    --learning_rate ${lr} \
-    --lr_scheduler_type ${lr_scheduler_type} \
-    --max_grad_norm ${max_grad_norm} \
-    --weight_decay ${weight_decay} \
-    --warmup_steps ${warmup_updates} \
-    --warmup_ratio ${warmup_ratio} \
-    --max_seq_length ${max_seq_length} \
-    --fp16 \
-    --logging_steps ${logging_steps} \
-    --save_total_limit 1 \
-    --evaluation_strategy ${eval_strategy} \
-    --save_strategy ${eval_strategy} \
-    --save_steps ${save_steps} \
-    --eval_steps ${save_steps} \
-    --load_best_model_at_end \
-    --report_to ${report_to} \
-    --run_name ${exp_name} \
-    --overwrite_output_dir \
-    --metric_for_best_model ${metric} \
-    --greater_is_better "True" \
-    --ddp_find_unused_parameter "False" \
-    --output_dir ${SAVE} \
-        2>&1 | tee ${SAVE}/log.txt
diff --git a/defect/run-adapter.sh b/defect/run-adapter.sh
deleted file mode 100644
index 354419f..0000000
--- a/defect/run-adapter.sh
+++ /dev/null
@@ -1,159 +0,0 @@
-#! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
-export HF_DATASETS_CACHE=checkpoints/hf_model
-export HF_METRICS_CACHE=checkpoints/hf_model
-
-cache_dir=${TRANSFORMERS_CACHE}
-
-
-metric="accuracy"
-export WANDB_PROJECT=glue
-export WANDB_WATCH="false"
-
-seed=42
-
-# ----- MAM adapter -----
-attn_mode="none"
-attn_option="none"
-attn_composition="add"
-attn_bn=16  # attn bottleneck dim
-
-ffn_mode="adapter"
-ffn_option="sequential"
-ffn_adapter_layernorm_option="none"
-ffn_adapter_init_option="lora"
-ffn_adapter_scalar="2"
-ffn_bn=16 # ffn bottleneck dim
-
-# ----- lora -----
-# attn_mode="lora"
-# attn_option="none"
-# attn_composition="add"
-# attn_bn=16
-
-# set ffn_mode to be 'lora' to use
-# lora at ffn as well
-
-# ffn_mode="lora"
-# ffn_option="none"
-# ffn_adapter_layernorm_option="none"
-# ffn_adapter_init_option="bert"
-# ffn_adapter_scalar="1"
-# ffn_bn=16
-
-lora_alpha=32
-lora_dropout=0.1
-lora_init="lora"
-
-
-# lora params are not set
-if [ -z ${lora_alpha+x} ];
-then
-    lora_alpha=0
-    lora_init="lora"
-    lora_dropout=0
-fi
-
-# set to 1 for debug mode which only
-# uses 1600 training examples
-debug=0
-
-# set to "wandb" to use weights & bias
-report_to="none"
-
-bsz=40
-gradient_steps=1
-
-# lr=5e-5
-lr=5e-5
-max_grad_norm=1
-# lr=1e-5
-# weight_decay=0
-weight_decay=0.1
-warmup_updates=0
-warmup_ratio=0.06
-max_steps=-1
-num_train_epochs=6
-max_tokens_per_batch=0
-max_seq_length=512
-
-lr_scheduler_type="polynomial"
-#metric=bleu
-unfreeze='ef_'
-max_eval_samples=1600
-logging_steps=1000
-
-eval_strategy="epoch"
-save_steps=5000
-
-# for seed in "${seed_list[@]}"; do
-model_name='codebert'
-exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
-exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
-exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
-exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
-exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}
-SAVE=checkpoints/${exp_name}
-
-if [ ! -d ${SAVE} ];then
-    mkdir -p ${SAVE}
-fi
-CUDA_VISIBLE_DEVICES=0 python -u run.py \
-    --model_name_or_path microsoft/codebert-baset \
-    --do_train \
-    --do_eval \
-    --do_predict \
-    --max_seq_length 512 \
-    --per_device_train_batch_size ${bsz} \
-    --per_device_eval_batch_size ${bsz} \
-    --max_tokens_per_batch ${max_tokens_per_batch} \
-    --adam_beta1 0.9 \
-    --adam_beta2 0.98 \
-    --adam_epsilon 1e-6 \
-    --lora_alpha ${lora_alpha} \
-    --lora_dropout ${lora_dropout} \
-    --lora_init ${lora_init} \
-    --attn_mode ${attn_mode} \
-    --attn_option ${attn_option} \
-    --attn_composition ${attn_composition} \
-    --ffn_mode ${ffn_mode} \
-    --ffn_option ${ffn_option} \
-    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
-    --ffn_adapter_scalar ${ffn_adapter_scalar} \
-    --ffn_adapter_init_option ${ffn_adapter_init_option} \
-    --mid_dim 800 \
-    --attn_bn ${attn_bn} \
-    --ffn_bn ${ffn_bn} \
-    --seed ${seed} \
-    --unfreeze_params ${unfreeze} \
-    --max_eval_samples ${max_eval_samples} \
-    --gradient_accumulation_steps ${gradient_steps} \
-    --max_steps ${max_steps} \
-    --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train.jsonl \
-    --valid_filename=dataset/valid.jsonl \
-    --test_filename=dataset/test.jsonl \
-    --learning_rate ${lr} \
-    --lr_scheduler_type ${lr_scheduler_type} \
-    --max_grad_norm ${max_grad_norm} \
-    --weight_decay ${weight_decay} \
-    --warmup_steps ${warmup_updates} \
-    --warmup_ratio ${warmup_ratio} \
-    --max_seq_length ${max_seq_length} \
-    --fp16 \
-    --logging_steps ${logging_steps} \
-    --save_total_limit 1 \
-    --evaluation_strategy ${eval_strategy} \
-    --save_strategy ${eval_strategy} \
-    --save_steps ${save_steps} \
-    --eval_steps ${save_steps} \
-    --load_best_model_at_end \
-    --report_to ${report_to} \
-    --run_name ${exp_name} \
-    --overwrite_output_dir \
-    --metric_for_best_model ${metric} \
-    --greater_is_better "True" \
-    --ddp_find_unused_parameter "False" \
-    --output_dir ${SAVE} \
-        2>&1 | tee ${SAVE}/log.txt
diff --git a/defect/run-lora.sh b/defect/run-lora.sh
index 2d29c34..166e8e8 100644
--- a/defect/run-lora.sh
+++ b/defect/run-lora.sh
@@ -1,10 +1,22 @@
 #! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
+#SBATCH --time=25:00:00          # Maximum run time in hh:mm:ss
+#SBATCH --mem=32000             # Maximum memory required (in megabytes)
+#SBATCH --job-name=default_479  # Job name (to track progress)
+#SBATCH --partition=gpu         # Partition on which to run job
+#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
+#SBATCH --constraint='gpu_32gb&gpu_v100'
+#SBATCH --licenses=common
+
+module load anaconda
+conda activate fine-tune-bert
+
+export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
+export TF_ENABLE_ONEDNN_OPTS=0
+export HF_HOME=checkpoints/hf_model
 export HF_DATASETS_CACHE=checkpoints/hf_model
 export HF_METRICS_CACHE=checkpoints/hf_model
 
-cache_dir=${TRANSFORMERS_CACHE}
+cache_dir=${HF_HOME}
 
 
 metric="f1"
@@ -101,7 +113,7 @@ if [ ! -d ${SAVE} ];then
 fi
 CUDA_VISIBLE_DEVICES=0 python -u run.py \
     --model_name_or_path microsoft/codebert-base \
-    --do_train False\
+    --do_train \
     --do_eval \
     --do_predict \
     --max_seq_length 512 \
diff --git a/defect/run-none.sh b/defect/run-none.sh
index a2d62cb..4b0fe5b 100644
--- a/defect/run-none.sh
+++ b/defect/run-none.sh
@@ -1,10 +1,22 @@
 #! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
+#SBATCH --time=25:00:00          # Maximum run time in hh:mm:ss
+#SBATCH --mem=32000             # Maximum memory required (in megabytes)
+#SBATCH --job-name=default_479  # Job name (to track progress)
+#SBATCH --partition=gpu         # Partition on which to run job
+#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
+#SBATCH --constraint='gpu_32gb&gpu_v100'
+#SBATCH --licenses=common
+
+module load anaconda
+conda activate fine-tune-bert
+
+export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
+export TF_ENABLE_ONEDNN_OPTS=0
+export HF_HOME=checkpoints/hf_model
 export HF_DATASETS_CACHE=checkpoints/hf_model
 export HF_METRICS_CACHE=checkpoints/hf_model
 
-cache_dir=${TRANSFORMERS_CACHE}
+cache_dir=${HF_HOME}
 
 
 metric="accuracy"
@@ -16,7 +28,7 @@ seed=42
 # ----- MAM adapter -----
 attn_mode="none"
 attn_option="none"
-attn_composition="none"
+attn_composition="add"
 attn_bn=16  # attn bottleneck dim
 
 ffn_mode="none"
@@ -74,7 +86,7 @@ weight_decay=0.1
 warmup_updates=0
 warmup_ratio=0.06
 max_steps=-1
-num_train_epochs=6
+num_train_epochs=5
 max_tokens_per_batch=0
 max_seq_length=512
 
@@ -82,6 +94,7 @@ lr_scheduler_type="polynomial"
 #metric=bleu
 unfreeze='ef_'
 max_eval_samples=1600
+max_train_samples=100
 logging_steps=1000
 
 eval_strategy="epoch"
@@ -127,6 +140,7 @@ CUDA_VISIBLE_DEVICES=0 python -u run.py \
     --ffn_bn ${ffn_bn} \
     --seed ${seed} \
     --unfreeze_params ${unfreeze} \
+    --max_train_samples ${max_train_samples} \
     --max_eval_samples ${max_eval_samples} \
     --gradient_accumulation_steps ${gradient_steps} \
     --max_steps ${max_steps} \
diff --git a/defect/run-parallel-adapter.sh b/defect/run-parallel-adapter.sh
index 09d0de9..8222597 100644
--- a/defect/run-parallel-adapter.sh
+++ b/defect/run-parallel-adapter.sh
@@ -1,10 +1,22 @@
 #! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
+#SBATCH --time=25:00:00          # Maximum run time in hh:mm:ss
+#SBATCH --mem=32000             # Maximum memory required (in megabytes)
+#SBATCH --job-name=default_479  # Job name (to track progress)
+#SBATCH --partition=gpu         # Partition on which to run job
+#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
+#SBATCH --constraint='gpu_32gb&gpu_v100'
+#SBATCH --licenses=common
+
+module load anaconda
+conda activate fine-tune-bert
+
+export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
+export TF_ENABLE_ONEDNN_OPTS=0
+export HF_HOME=checkpoints/hf_model
 export HF_DATASETS_CACHE=checkpoints/hf_model
 export HF_METRICS_CACHE=checkpoints/hf_model
 
-cache_dir=${TRANSFORMERS_CACHE}
+cache_dir=${HF_HOME}
 
 
 metric="accuracy"
@@ -16,7 +28,7 @@ seed=42
 # ----- MAM adapter -----
 attn_mode="none"
 attn_option="none"
-attn_composition="none"
+attn_composition="add"
 attn_bn=16  # attn bottleneck dim
 
 ffn_mode="adapter"
diff --git a/defect/run-prefix.sh b/defect/run-prefix.sh
deleted file mode 100644
index 6cec2cb..0000000
--- a/defect/run-prefix.sh
+++ /dev/null
@@ -1,159 +0,0 @@
-#! /bin/bash
-
-export TRANSFORMERS_CACHE=checkpoints/hf_model
-export HF_DATASETS_CACHE=checkpoints/hf_model
-export HF_METRICS_CACHE=checkpoints/hf_model
-
-cache_dir=${TRANSFORMERS_CACHE}
-
-
-metric="accuracy"
-export WANDB_PROJECT=glue
-export WANDB_WATCH="false"
-
-seed=42
-
-# ----- MAM adapter -----
-attn_mode="prefix"
-attn_option="concat"
-attn_composition="add"
-attn_bn=16  # attn bottleneck dim
-
-ffn_mode="none"
-ffn_option="none"
-ffn_adapter_layernorm_option="none"
-ffn_adapter_init_option="lora"
-ffn_adapter_scalar="2"
-ffn_bn=16 # ffn bottleneck dim
-
-# ----- lora -----
-# attn_mode="lora"
-# attn_option="none"
-# attn_composition="add"
-# attn_bn=16
-
-# set ffn_mode to be 'lora' to use
-# lora at ffn as well
-
-# ffn_mode="lora"
-# ffn_option="none"
-# ffn_adapter_layernorm_option="none"
-# ffn_adapter_init_option="bert"
-# ffn_adapter_scalar="1"
-# ffn_bn=16
-
-# lora_alpha=32
-# lora_dropout=0.1
-# lora_init="lora"
-
-
-# lora params are not set
-if [ -z ${lora_alpha+x} ];
-then
-    lora_alpha=0
-    lora_init="lora"
-    lora_dropout=0
-fi
-
-# set to 1 for debug mode which only
-# uses 1600 training examples
-debug=0
-
-# set to "wandb" to use weights & bias
-report_to="none"
-
-bsz=40
-gradient_steps=1
-
-# lr=5e-5
-lr=5e-5
-max_grad_norm=1
-# lr=1e-5
-# weight_decay=0
-weight_decay=0.1
-warmup_updates=0
-warmup_ratio=0.06
-max_steps=-1
-num_train_epochs=6
-max_tokens_per_batch=0
-max_seq_length=512
-
-lr_scheduler_type="polynomial"
-#metric=bleu
-unfreeze='ef_'
-max_eval_samples=1600
-logging_steps=1000
-
-eval_strategy="epoch"
-save_steps=5000
-
-# for seed in "${seed_list[@]}"; do
-model_name='codebert'
-exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
-exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
-exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
-exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
-exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}_tes4
-SAVE=checkpoints/${exp_name}
-
-if [ ! -d ${SAVE} ];then
-    mkdir -p ${SAVE}
-fi
-CUDA_VISIBLE_DEVICES=0 python -u run.py \
-    --model_name_or_path microsoft/codebert-base \
-    --do_train \
-    --do_eval \
-    --do_predict \
-    --max_seq_length 512 \
-    --per_device_train_batch_size ${bsz} \
-    --per_device_eval_batch_size ${bsz} \
-    --max_tokens_per_batch ${max_tokens_per_batch} \
-    --adam_beta1 0.9 \
-    --adam_beta2 0.98 \
-    --adam_epsilon 1e-6 \
-    --lora_alpha ${lora_alpha} \
-    --lora_dropout ${lora_dropout} \
-    --lora_init ${lora_init} \
-    --attn_mode ${attn_mode} \
-    --attn_option ${attn_option} \
-    --attn_composition ${attn_composition} \
-    --ffn_mode ${ffn_mode} \
-    --ffn_option ${ffn_option} \
-    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
-    --ffn_adapter_scalar ${ffn_adapter_scalar} \
-    --ffn_adapter_init_option ${ffn_adapter_init_option} \
-    --mid_dim 400 \
-    --attn_bn ${attn_bn} \
-    --ffn_bn ${ffn_bn} \
-    --seed ${seed} \
-    --unfreeze_params ${unfreeze} \
-    --max_eval_samples ${max_eval_samples} \
-    --gradient_accumulation_steps ${gradient_steps} \
-    --max_steps ${max_steps} \
-    --num_train_epochs ${num_train_epochs} \
-    --train_filename=dataset/train.jsonl \
-    --valid_filename=dataset/valid.jsonl \
-    --test_filename=dataset/test.jsonl \
-    --learning_rate ${lr} \
-    --lr_scheduler_type ${lr_scheduler_type} \
-    --max_grad_norm ${max_grad_norm} \
-    --weight_decay ${weight_decay} \
-    --warmup_steps ${warmup_updates} \
-    --warmup_ratio ${warmup_ratio} \
-    --max_seq_length ${max_seq_length} \
-    --fp16 \
-    --logging_steps ${logging_steps} \
-    --save_total_limit 1 \
-    --evaluation_strategy ${eval_strategy} \
-    --save_strategy ${eval_strategy} \
-    --save_steps ${save_steps} \
-    --eval_steps ${save_steps} \
-    --load_best_model_at_end \
-    --report_to ${report_to} \
-    --run_name ${exp_name} \
-    --overwrite_output_dir \
-    --metric_for_best_model ${metric} \
-    --greater_is_better "True" \
-    --ddp_find_unused_parameter "False" \
-    --output_dir ${SAVE} \
-        2>&1 | tee ${SAVE}/log.txt
diff --git a/defect/run.py b/defect/run.py
index ee24f66..fefa29b 100644
--- a/defect/run.py
+++ b/defect/run.py
@@ -25,6 +25,7 @@ import numpy as np
 from utils import CodeDataset, get_last_checkpoint
 from sklearn.metrics import recall_score, precision_score, f1_score
 import transformers
+from transformers import BertConfig
 from transformers import (
     AutoTokenizer,
     HfArgumentParser,
@@ -199,12 +200,26 @@ def main():
     # Set seed before initializing model.
     set_seed(training_args.seed)
     last_checkpoint = get_last_checkpoint(training_args.output_dir)
-    config = RobertaConfig.from_pretrained(
-        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+    # config = RobertaConfig.from_pretrained(
+    #     model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+    #     num_labels=2
+    # )
+    # if training_args.do_predict and not training_args.do_train:
+    #     config = RobertaConfig.from_pretrained(
+    #         last_checkpoint,
+    #         num_labels=2
+    #     )
+    
+    print("Checkpoint:", last_checkpoint)
+    config = BertConfig.from_pretrained(
+        'microsoft/codebert-base',
         num_labels=2
     )
+
     if training_args.do_predict and not training_args.do_train:
-        config = RobertaConfig.from_pretrained(
+        print('Configuration for prediction...')
+        config = BertConfig.from_pretrained(
+            'microsoft/codebert-base',
             last_checkpoint,
             num_labels=2
         )
@@ -223,6 +238,8 @@ def main():
         use_fast=model_args.use_fast_tokenizer,
         revision=model_args.model_revision,
         use_auth_token=True if model_args.use_auth_token else None,
+        token="Bearer <REDACTED-HF-TOKEN>"
+
     )
     model = RobertaForSequenceClassification.from_pretrained(
         model_args.model_name_or_path,
@@ -231,6 +248,8 @@ def main():
         cache_dir=model_args.cache_dir,
         revision=model_args.model_revision,
         use_auth_token=True if model_args.use_auth_token else None,
+        token="Bearer <REDACTED-HF-TOKEN>"
+
     )
 
     dataset_loader = CodeDataset(os.path.dirname(model_args.train_filename))
@@ -242,13 +261,15 @@ def main():
 
     def preprocess_function(examples):
         # Tokenize the texts
-        args = ((examples['code1'], examples['code2']))
+        # args = examples['code']
 
         # {input_ids, attention_mask}
-        result = tokenizer(*args, padding="max_length", max_length=max_seq_length, truncation='longest_first')
+        # result = tokenizer(*args, padding="max_length", max_length=max_seq_length, truncation='longest_first')
+        result = tokenizer(examples['code'], padding="max_length", max_length=max_seq_length, truncation=True)
         # Map labels to IDs (not necessary for GLUE tasks)
         if label_to_id is not None and "label" in examples:
-            result["label"] = [(label_to_id[l] if l != -1 else -1) for l in examples["label"]]
+            # result["label"] = [(label_to_id[l] if l != -1 else -1) for l in examples["label"]]
+            result["labels"] = [label_to_id.get(l, -1) for l in examples["label"]]
         return result
 
     if tune_args.attn_mode != "none" or tune_args.ffn_mode != "none":
@@ -326,6 +347,7 @@ def main():
 
             logger.info("load model state from %s", last_checkpoint)
             model.load_state_dict(torch.load(os.path.join(last_checkpoint, "pytorch_model.bin")))
+            # model.load_state_dict(torch.load(os.path.join('checkpoints/hf_model/hub/models--microsoft--codebert-base/snapshots/3b0952feddeffad0063f274080e3c23d75e7eb39/', "pytorch_model.bin")), strict=False)
             # Initialize our Trainer
             trainer = Trainer(
                 model=model,
diff --git a/defect/utils.py b/defect/utils.py
index 9975050..71bf8b2 100644
--- a/defect/utils.py
+++ b/defect/utils.py
@@ -52,6 +52,9 @@ def read_dataset(filename):
 
 
 class CodeDataset:
+    def __init__(self, data_dir):
+        self.data_dir = data_dir
+        
     def load(self, file_path, data_num=-1):
         dataset_dict = {'code': [], 'label': []}
         count = 0
diff --git a/petl/petl_enc_model.py b/petl/petl_enc_model.py
index 7503fed..4fcbad9 100644
--- a/petl/petl_enc_model.py
+++ b/petl/petl_enc_model.py
@@ -1,4 +1,4 @@
-from transformers import PreTrainedModel, RobertaConfig
+from transformers import PreTrainedModel, RobertaConfig, BertConfig
 from petl.petl_factory import Prefix, MLP_Bias, Bias, PrefixDirectInit
 from transformers.utils import logging
 
@@ -11,7 +11,7 @@ class PETLEncModel(PreTrainedModel):
         self.args = args
         self.pretrained_model = pretrained_model
 
-        if isinstance(config, RobertaConfig):
+        if isinstance(config, RobertaConfig) or isinstance(config, BertConfig):
             self.match_n_layer = config.num_hidden_layers
             self.match_n_head = config.num_attention_heads
             self.n_embd = config.hidden_size

=== NEW FILE: clone/run-bitfit.sh ===
#! /bin/bash
#SBATCH --time=15:00:00          # Maximum run time in hh:mm:ss
#SBATCH --mem=32000             # Maximum memory required (in megabytes)
#SBATCH --job-name=default_479  # Job name (to track progress)
#SBATCH --partition=gpu         # Partition on which to run job
#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
#SBATCH --constraint=gpu_v100
#SBATCH --licenses=common

module load anaconda
conda activate fine-tune-bert

export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
export TF_ENABLE_ONEDNN_OPTS=0
export HF_HOME=checkpoints/hf_model
export HF_DATASETS_CACHE=checkpoints/hf_model
export HF_METRICS_CACHE=checkpoints/hf_model

cache_dir=${HF_HOME}


metric="accuracy"
export WANDB_PROJECT=glue
export WANDB_WATCH="false"

seed=42

# ----- MAM adapter -----
attn_mode="bitfit"
attn_option="none"
attn_composition="add"
attn_bn=16  # attn bottleneck dim

ffn_mode="none"
ffn_option="none"
ffn_adapter_layernorm_option="none"
ffn_adapter_init_option="lora"
ffn_adapter_scalar="2"
ffn_bn=16 # ffn bottleneck dim

# ----- lora -----
# attn_mode="lora"
# attn_option="none"
# attn_composition="add"
# attn_bn=16

# set ffn_mode to be 'lora' to use
# lora at ffn as well

# ffn_mode="lora"
# ffn_option="none"
# ffn_adapter_layernorm_option="none"
# ffn_adapter_init_option="bert"
# ffn_adapter_scalar="1"
# ffn_bn=16

lora_alpha=32
lora_dropout=0.1
lora_init="lora"


# lora params are not set
if [ -z ${lora_alpha+x} ];
then
    lora_alpha=0
    lora_init="lora"
    lora_dropout=0
fi

# set to 1 for debug mode which only
# uses 1600 training examples
debug=0

# set to "wandb" to use weights & bias
report_to="none"

bsz=40
gradient_steps=1

# lr=5e-5
lr=1e-4
max_grad_norm=1
# lr=1e-5
# weight_decay=0
weight_decay=0.1
warmup_updates=0
warmup_ratio=0.06
max_steps=-1
num_train_epochs=10
max_tokens_per_batch=0
max_seq_length=512

lr_scheduler_type="polynomial"
#metric=bleu
unfreeze='ef_'
max_eval_samples=1600
max_train_samples=100
logging_steps=1000

eval_strategy="epoch"
save_steps=5000

# for seed in "${seed_list[@]}"; do
model_name='codebert'
exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}
SAVE=checkpoints/${exp_name}

if [ ! -d ${SAVE} ];then
    mkdir -p ${SAVE}
fi
CUDA_VISIBLE_DEVICES=0 python -u run.py \
    --model_name_or_path microsoft/codebert-base \
    --do_train \
    --do_eval \
    --do_predict \
    --max_seq_length 512 \
    --per_device_train_batch_size ${bsz} \
    --per_device_eval_batch_size ${bsz} \
    --max_tokens_per_batch ${max_tokens_per_batch} \
    --adam_beta1 0.9 \
    --adam_beta2 0.98 \
    --adam_epsilon 1e-6 \
    --lora_alpha ${lora_alpha} \
    --lora_dropout ${lora_dropout} \
    --lora_init ${lora_init} \
    --attn_mode ${attn_mode} \
    --attn_option ${attn_option} \
    --attn_composition ${attn_composition} \
    --ffn_mode ${ffn_mode} \
    --ffn_option ${ffn_option} \
    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
    --ffn_adapter_scalar ${ffn_adapter_scalar} \
    --ffn_adapter_init_option ${ffn_adapter_init_option} \
    --mid_dim 800 \
    --attn_bn ${attn_bn} \
    --ffn_bn ${ffn_bn} \
    --seed ${seed} \
    --unfreeze_params ${unfreeze} \
    --max_train_samples ${max_train_samples} \
    --max_eval_samples ${max_eval_samples} \
    --gradient_accumulation_steps ${gradient_steps} \
    --max_steps ${max_steps} \
    --num_train_epochs ${num_train_epochs} \
    --train_filename=dataset/train.txt \
    --valid_filename=dataset/valid.txt \
    --test_filename=dataset/test.txt \
    --learning_rate ${lr} \
    --lr_scheduler_type ${lr_scheduler_type} \
    --max_grad_norm ${max_grad_norm} \
    --weight_decay ${weight_decay} \
    --warmup_steps ${warmup_updates} \
    --warmup_ratio ${warmup_ratio} \
    --max_seq_length ${max_seq_length} \
    --fp16 \
    --logging_steps ${logging_steps} \
    --save_total_limit 1 \
    --evaluation_strategy ${eval_strategy} \
    --save_strategy ${eval_strategy} \
    --save_steps ${save_steps} \
    --eval_steps ${save_steps} \
    --load_best_model_at_end \
    --report_to ${report_to} \
    --run_name ${exp_name} \
    --overwrite_output_dir \
    --metric_for_best_model ${metric} \
    --greater_is_better "True" \
    --ddp_find_unused_parameter "False" \
    --output_dir ${SAVE}
=== NEW FILE: defect/run-bitfit.sh ===
#! /bin/bash
#SBATCH --time=25:00:00          # Maximum run time in hh:mm:ss
#SBATCH --mem=32000             # Maximum memory required (in megabytes)
#SBATCH --job-name=default_479  # Job name (to track progress)
#SBATCH --partition=gpu         # Partition on which to run job
#SBATCH --gres=gpu:1            # Don't change this, it requests a GPU
#SBATCH --constraint='gpu_32gb&gpu_v100'
#SBATCH --licenses=common

module load anaconda
conda activate fine-tune-bert

export LD_LIBRARY_PATH=/home/cse479/spatel20/.conda/envs/fine-tune-bert/lib/:$LD_LIBRARY_PATH
export TF_ENABLE_ONEDNN_OPTS=0
export HF_HOME=checkpoints/hf_model
export HF_DATASETS_CACHE=checkpoints/hf_model
export HF_METRICS_CACHE=checkpoints/hf_model

cache_dir=${HF_HOME}


metric="accuracy"
export WANDB_PROJECT=glue
export WANDB_WATCH="false"

seed=42

# ----- MAM adapter -----
attn_mode="bitfit"
attn_option="none"
attn_composition="add"
attn_bn=16  # attn bottleneck dim

ffn_mode="none"
ffn_option="none"
ffn_adapter_layernorm_option="none"
ffn_adapter_init_option="lora"
ffn_adapter_scalar="2"
ffn_bn=16 # ffn bottleneck dim

# ----- lora -----
# attn_mode="lora"
# attn_option="none"
# attn_composition="add"
# attn_bn=16

# set ffn_mode to be 'lora' to use
# lora at ffn as well

# ffn_mode="lora"
# ffn_option="none"
# ffn_adapter_layernorm_option="none"
# ffn_adapter_init_option="bert"
# ffn_adapter_scalar="1"
# ffn_bn=16

# lora_alpha=32
# lora_dropout=0.1
# lora_init="lora"


# lora params are not set
if [ -z ${lora_alpha+x} ];
then
    lora_alpha=0
    lora_init="lora"
    lora_dropout=0
fi

# set to 1 for debug mode which only
# uses 1600 training examples
debug=0

# set to "wandb" to use weights & bias
report_to="none"

bsz=40
gradient_steps=1

# lr=5e-5
lr=5e-5
max_grad_norm=1
# lr=1e-5
# weight_decay=0
weight_decay=0.1
warmup_updates=0
warmup_ratio=0.06
max_steps=-1
num_train_epochs=5
max_tokens_per_batch=0
max_seq_length=512

lr_scheduler_type="polynomial"
#metric=bleu
unfreeze='ef_'
max_eval_samples=1600
max_train_samples=100
logging_steps=1000

eval_strategy="epoch"
save_steps=5000

# for seed in "${seed_list[@]}"; do
model_name='codebert'
exp_name=${model_name}.am_${attn_mode}.ao_${attn_option}.fm_${ffn_mode}
exp_name+=.fo_${ffn_option}.abn${preseqlen}.fbn${ffn_bn_len}.ac_${attn_composition}
exp_name+=.fl_${ffn_adapter_layernorm_option}.finit_${ffn_adapter_init_option}
exp_name+=.fs_${ffn_adapter_scalar}.unfrz_${unfreeze}.ne${num_train_epochs}
exp_name+=.warm${warmup_ratio}.wd${weight_decay}.seed${seed}
SAVE=checkpoints/${exp_name}

if [ ! -d ${SAVE} ];then
    mkdir -p ${SAVE}
fi
CUDA_VISIBLE_DEVICES=0 python -u run.py \
    --model_name_or_path microsoft/codebert-base\
    --do_train \
    --do_eval \
    --do_predict \
    --max_seq_length 512 \
    --per_device_train_batch_size ${bsz} \
    --per_device_eval_batch_size ${bsz} \
    --max_tokens_per_batch ${max_tokens_per_batch} \
    --adam_beta1 0.9 \
    --adam_beta2 0.98 \
    --adam_epsilon 1e-6 \
    --lora_alpha ${lora_alpha} \
    --lora_dropout ${lora_dropout} \
    --lora_init ${lora_init} \
    --attn_mode ${attn_mode} \
    --attn_option ${attn_option} \
    --attn_composition ${attn_composition} \
    --ffn_mode ${ffn_mode} \
    --ffn_option ${ffn_option} \
    --ffn_adapter_layernorm_option ${ffn_adapter_layernorm_option} \
    --ffn_adapter_scalar ${ffn_adapter_scalar} \
    --ffn_adapter_init_option ${ffn_adapter_init_option} \
    --mid_dim 800 \
    --attn_bn ${attn_bn} \
    --ffn_bn ${ffn_bn} \
    --seed ${seed} \
    --unfreeze_params ${unfreeze} \
    --max_train_samples ${max_train_samples} \
    --max_eval_samples ${max_eval_samples} \
    --gradient_accumulation_steps ${gradient_steps} \
    --max_steps ${max_steps} \
    --num_train_epochs ${num_train_epochs} \
    --train_filename=dataset/train.jsonl \
    --valid_filename=dataset/valid.jsonl \
    --test_filename=dataset/test.jsonl \
    --learning_rate ${lr} \
    --lr_scheduler_type ${lr_scheduler_type} \
    --max_grad_norm ${max_grad_norm} \
    --weight_decay ${weight_decay} \
    --warmup_steps ${warmup_updates} \
    --warmup_ratio ${warmup_ratio} \
    --max_seq_length ${max_seq_length} \
    --fp16 \
    --logging_steps ${logging_steps} \
    --save_total_limit 1 \
    --evaluation_strategy ${eval_strategy} \
    --save_strategy ${eval_strategy} \
    --save_steps ${save_steps} \
    --eval_steps ${save_steps} \
    --load_best_model_at_end \
    --report_to ${report_to} \
    --run_name ${exp_name} \
    --overwrite_output_dir \
    --metric_for_best_model ${metric} \
    --greater_is_better "True" \
    --ddp_find_unused_parameter "False" \
    --output_dir ${SAVE} \
        2>&1 | tee ${SAVE}/log.txt
